src/backend.ts

```diff
@@ -311,14 +321,69 @@ export function setupSuspendResumeHandler(): () => void {
 export function setupFocusChangeHandler(): () => void {
   let appIsStartingUp: boolean = false;
   let lastPid = 0;
+  let lastAppid = 0;
+  let validKeyEvent: SystemKeyEvent | null = null;
+  let keyEventFunction = (e: SystemKeyEvent) => {
+    const cancelDebouncedEvent = debounce(() => {
+      validKeyEvent = null;
+    }, 1000);
+    if (e.eKey === 0) {
+      // Have not found any race condition issues with this approach since the key event fires long before focus change
+      validKeyEvent = e;
+      cancelDebouncedEvent();
+    }
+  };
+  const unregisterSystemKeyEvents = () => {
+    keyEventFunction = () => {};
+  };
+  SteamClient.System.UI.RegisterForSystemKeyEvents((e) => keyEventFunction(e));
+
   const { unregister: unregisterFocusChangeEvents } =
     SteamClient.System.UI.RegisterForFocusChangeEvents(
       throttle(async (fce: FocusChangeEvent) => {
         // don't try anything while an application is launching or it could pause it midlaunch
         if (appIsStartingUp) return;
         // skip if we already got such an event before
-        if (fce.focusedApp.pid === lastPid) return;
+        if (
+          fce.focusedApp.pid === lastPid &&
+          fce.focusedApp.appid === lastAppid
+        )
+          return;
         lastPid = fce.focusedApp.pid;
+        lastAppid = fce.focusedApp.appid;
+        // Only pause if the user has enabled it
+        if (
+          fce.focusedApp.appid === 769 &&
+          fce.focusedApp.strExeName !== "steamwebhelper" &&
```

Comment:
I wonder if the test for not "steamwebhelper" is really desirable here ? For instance, when I have an app running but I go to my Home screen `strExeName` will be "steamwebhelper" but I would still want the "Pause on overlay" feature to apply. Just checking if `fce.focusedApp.appid` is `769` might be enough.

Reply:
Tested a bit more and it all seems to work well with just `fce.focusedApp.appid === 769`, no `fce.focusedApp.strExeName !== "steamwebhelper"`.